app/v3/controllers/requestParsers/validators/SubmitSelfEmploymentBsasValidator.scala

```diff
@@ -52,78 +57,174 @@ class SubmitSelfEmploymentBsasValidator extends Validator[SubmitSelfEmploymentBs
+        path = "/additions/paymentsToSubcontractorsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.wagesAndStaffCostsDisallowable,
+        path = "/additions/wagesAndStaffCostsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.carVanTravelExpensesDisallowable,
+        path = "/additions/carVanTravelExpensesDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.premisesRunningCostsDisallowable,
+        path = "/additions/premisesRunningCostsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.maintenanceCostsDisallowable,
+        path = "/additions/maintenanceCostsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.adminCostsDisallowable,
+        path = "/additions/adminCostsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.interestOnBankOtherLoansDisallowable,
+        path = "/additions/interestOnBankOtherLoansDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.financeChargesDisallowable,
+        path = "/additions/financeChargesDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.irrecoverableDebtsDisallowable,
+        path = "/additions/irrecoverableDebtsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.professionalFeesDisallowable,
+        path = "/additions/professionalFeesDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.depreciationDisallowable,
+        path = "/additions/depreciationDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.otherExpensesDisallowable,
+        path = "/additions/otherExpensesDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.advertisingCostsDisallowable,
+        path = "/additions/advertisingCostsDisallowable",
+      ),
+      NumberValidation.validateAdjustment(
+        field = additions.businessEntertainmentCostsDisallowable,
+        path = "/additions/businessEntertainmentCostsDisallowable",
+      )
+    ).flatten
+  }
+
+  private def bothExpensesValidation: SubmitSelfEmploymentBsasRawData => List[List[MtdError]] = { data =>
     val model: SubmitSelfEmploymentBsasRequestBody = data.body.json.as[SubmitSelfEmploymentBsasRequestBody]
 
     List(
       if(model.expenses.exists(_.isBothSupplied)
-        || (!(model.additions.isEmpty || model.additions.exists(_.isEmpty))
-        && (!(model.expenses.isEmpty || model.expenses.exists(_.isConsolidatedExpensesEmpty))))){
+        || (!(model.expenses.isEmpty || model.expenses.exists(_.isConsolidatedExpensesEmpty))
+        && (!(model.additions.isEmpty || model.additions.exists(_.isEmpty))))){
```

Comment:
It is not obvious that this is correct and there do not appear to be test cases that exercise the second two lines (comment them out and the tests still pass).

I would write some extra tests first (maybe move the code to the existing BothExpensesValidation/BothExpensesValidationSpec where testing exhaustively will prob be easier) but you might be able to simplify the logic to something like:
```scala
      if (model.expenses.exists(_.isBothSupplied)
          || (
            model.expenses.exists(x => x.consolidatedExpenses.isDefined) &&
            model.additions.exists(x => !x.isEmpty)
          )) { ...
```